src/controllers/Social.ts

```diff
@@ -92,15 +98,37 @@ export default class Social {
 
   private async _storageBalanceOf({
     accountId,
-    signer,
+    rpcURL,
   }: IStorageBalanceOfOptions): Promise<ISocialDBContractStorageBalance | null> {
-    return await signer.viewFunction({
+    const result = await viewFunction({
+      contractId: this.contractId,
+      method: ViewMethodEnum.StorageBalanceOf,
       args: {
         account_id: accountId,
       },
-      contractId: this.contractId,
-      methodName: ViewMethodEnum.StorageBalanceOf,
+      rpcURL,
     });
+
+    if (this.isStorageBalance(result)) {
+      return result;
+    } else if (result === null) {
+      return null;
+    } else {
+      throw new Error('Unexpected response format from storage_balance_of');
+    }
+  }
+
+  private isStorageBalance(
```

Comment:
To conform to the other function naming, private methods should be prefixed with an underscore: `_isStorageBalance`